api/src/service/domain/workflow/workflowitem_assign.ts

```diff
@@ -48,15 +51,47 @@ export async function assignWorkflowitem(
     workflowitemId,
     assignee,
   );
+
   if (Result.isErr(assignEvent)) {
     return new VError(assignEvent, "failed to create event");
   }
 
+  let permissionsGranted;
+  let permissionsRevoked;
+
+  if (workflowitem.type === "restricted") {
```

Comment:
check if requesting user is permitted to grant/revoke permissions
only grant/revoke write/admin permissions - view permissions should be untouched
pay attention to the order of events. e.g. The last event must be the user's revoke "workflowitem revoke permission".